src/libstd_unicode/lossy.rs

```diff
@@ -0,0 +1,154 @@
+use core::str as core_str;
+use core::fmt;
+use core::fmt::Write;
+use char;
+
+/// Iterator over lossy UTF-8 string
+#[unstable(feature = "str_internals", issue = "0")]
+pub struct Utf8LossyIter<'a> {
```

Comment:
can we write some more unit tests for this? it seems like quite a big chunk of code, and one for which it'd be easy to write some targeted unit tests...